cron.go

```diff
@@ -414,14 +476,20 @@ func (c *Cron) run(ctx context.Context) {
 				e.tick(effective)
 
 				go func(ctx context.Context, e *Entry, next time.Time) {
+					if e.mustDelete.Load() || e.Job.expired(effective) {
```

Comment:
The deletion does not need to happen exactly at the TTL time, we can keep retrying too, as long as the job is not triggered.